BreadCrumb Control/BreadCrumb Control/BreadCrumb.swift

```diff
@@ -181,66 +183,73 @@ public class CBreadcrumbControl: UIScrollView {
         return false
     }
     
-    func initialSetup( refresh: Bool) {
+    func initialSetup(refresh: Bool) {
         var changeRoot: Int = 0
-        if ((visibleRootButton) && (self.startButton == nil)) {
+        if visibleRootButton && self.startButton == nil {
             self.startButton = self.startRootButton()
             changeRoot = 1
-        } else if ((visibleRootButton == false) && (self.startButton != nil)){
+        } else if visibleRootButton == false && self.startButton != nil {
             changeRoot = 2
         }
-        if (self.containerView == nil ) {
-            let rectContainerView: CGRect = CGRect(origin: CGPoint(x: kStartButtonWidth+1, y: 0),
-                                                   size: CGSize(width: self.bounds.size.width - (kStartButtonWidth+1), height: self.frame.size.height))
-            self.containerView = UIView(frame:rectContainerView)
-            self.containerView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
-            self.clipsToBounds = true
+        if self.containerView == nil {
+            let rectContainerView = CGRect(origin: CGPoint(x: kStartButtonWidth + 1, y: 0),
+                                           size: CGSize(width: self.bounds.size.width - (kStartButtonWidth + 1), height: self.frame.size.height))
+            let containerView = UIView(frame:rectContainerView)
+            containerView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+            clipsToBounds = true
+            
+            self.addSubview(containerView)
             
-            self.addSubview( self.containerView)
+            self.containerView = containerView
         }
+        guard let containerView = self.containerView else { return }
 
-        self.containerView.backgroundColor = backgroundBCColor  //UIColor.white
+        containerView.backgroundColor = backgroundBCColor  //UIColor.white
         
-        if ((visibleRootButton) && (self.startButton != nil)) {
-            self.startButton.backgroundColor = backgroundRootButtonColor
+        if let startButton = self.startButton, visibleRootButton {
+            startButton.backgroundColor = backgroundRootButtonColor
         }
         
-        if (changeRoot == 1) {
-            self.addSubview( self.startButton)
-            let rectContainerView: CGRect = CGRect(origin: CGPoint(x: kStartButtonWidth+1, y: 0),
-                                                   size: CGSize(width: self.bounds.size.width - (kStartButtonWidth+1), height: self.frame.size.height))
-            self.containerView.frame = rectContainerView
-        } else if (changeRoot == 2) {
-            self.startButton.removeFromSuperview()
+        if changeRoot == 1 {
```

Comment:
If you can, I think it would be best to refactor these magic numbers into an `enum RootState`.

Reply:
Exactly. Magic numbers are not good. I'll think anything idea.